File: back_nestjs/src/data/data-base-fake.ts
/* eslint-disable prettier/prettier */
import { Injectable, Scope } from '@nestjs/common';
import { SubscriptionAddDto } from 'src/dtos/subscription-add.dto';
import { SubscriptionDto } from 'src/dtos/subscription.dto';

@Injectable({ scope: Scope.DEFAULT })
export class DataBaseFake {
  private static _subscriptions: SubscriptionDto[] = [];

  addSubscription = (
    subscription: SubscriptionAddDto,
    key: string,
  ): SubscriptionDto => {
    const indexExistsItem = DataBaseFake._subscriptions.findIndex(
      (x) => x.key == key,
    );
    const newSubscription = {
      ...subscription,
      key,
    };

    if (indexExistsItem > 0)
      DataBaseFake._subscriptions.splice(indexExistsItem, 1);
    DataBaseFake._subscriptions.push(newSubscription);

    return newSubscription;
  };

  getSubscription = (key: string): SubscriptionDto => {
    return DataBaseFake._subscriptions.find((x) => x.key == key);
  };

  getAllKeys = (): string[] => {
    return DataBaseFake._subscriptions.map((x) => x.key);
  };
}

File: backend_net/backend_net/Fakes/DatabaseFake.cs
using backend_net.Dtos;

namespace backend_net.Fakes
{
    public static class DatabaseFake
    {
        public static List<SubscriptionDto> _subscriptions = new();

        public static void AddSubscription(string key, SubscriptionAddDto subscription)
        {
            _subscriptions.RemoveAll(x => x.key == key);
            _subscriptions.Add(new() { key = key, Subscription = subscription });
        }
        public static IReadOnlyList<SubscriptionDto> GetAll() => _subscriptions;
        public static SubscriptionDto? GetByKey(string key) => _subscriptions.FirstOrDefault(x => x.key == key);
    }
}
